document/CoreFormat.mdk

```diff
@@ -491,22 +533,57 @@ This allows tools to include information about the themselves that is not explic
 A `file` object represents a single file. In some cases, this file might be nested within another file,
 referred to as the "containing file."
 
+`file` objects occur as members of arrays within the `run.files` object (&sect;[#run-files]). 
+
 If the `file` object does not represent a nested file, neither of the properties
-`pathFromParent` (&sect;[#file-pathFromParent]) or `offsetFromParent` (&sect;[#file-offsetFromParent]) shall be present.
+`uri` (&sect;[#file-uri]) or `offsetFromParent` (&sect;[#file-offsetFromParent]) shall be present.
 
 If the location of a nested file can be expressed either by means of a path within the containing file
-or by means of a byte offset from the start of the containing file, then either `pathFromParent`
+or by means of a byte offset from the start of the containing file, then either `uri`
 or `offsetFromParent` (&sect;[#file-offsetFromParent]) or both shall be present; they must not both be absent.
 
-### `pathFromParent` property { #file-pathFromParent }
+### `uri` property { #file-uri }
 
 If the location of a nested file with respect to its containing file can be expressed
 only by means of a path within its containing file,
```

Comment:
If you can express a URI, you should express the URI, in my opinion. There's no clear winner here: if a producer doesn't emit it, that complicates the life of a consumer. If you can provide an offset in this case as well, ok, fine, why prohibit that. It would be rare, as you say.